x/claim/keeper/claim.go

```diff
@@ -0,0 +1,235 @@
+package keeper
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/arkeonetwork/arkeo/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/proto"
+)
+
+// SetClaimRecord sets a claim record for an address in store
+func (k Keeper) SetClaimRecord(ctx sdk.Context, claimRecord types.ClaimRecord) error {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, chainToStorePrefix(claimRecord.Chain))
+
+	bz, err := proto.Marshal(&claimRecord)
+	if err != nil {
+		return err
+	}
+
+	addr := []byte(strings.ToLower(claimRecord.Address))
+
+	// validate address if valid based on chain
+	switch claimRecord.Chain {
```

Comment:
move these few lines of validation related logic to the top , before marshal the object will be more efficient. because if the address is invalid , you don't need to pay the cost to marshal it.

Reply:
[55343a6](https://github.com/arkeonetwork/arkeo/pull/29/commits/55343a63b8648e68174d54cf63eaf9524c965e4e)